utils/sampler/rand.go

```diff
@@ -37,9 +37,9 @@ type rng struct {
 
 // Seed uses the provided seed value to initialize the generator to a
 // deterministic state.
-func (r *rng) Seed(seed int64) {
+func (r *rng) Seed(seed uint64) {
 	r.lock.Lock()
-	r.rng.Seed(uint64(seed))
+	r.rng.Seed(seed)
```

Comment:
We really shouldn't have been casting this before... There wasn't a good reason to take in an `int64`.